tools/cli/table.go

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2022 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/olekukonko/tablewriter"
+
+	"github.com/uber/cadence/common/types"
+)
+
+// TableOptions allows passing optional flags for altering rendered table
+type TableOptions struct {
+	// OptionalColumns may contain column header names which can be hidden
+	OptionalColumns map[string]bool
+
+	// Border specified whether to render table border
+	Border bool
+
+	// NoColor will not use color characters while printing table
+	NoColor bool
+
+	// PrintRawTime will print time as int64 unix nanos
+	PrintRawTime bool
+	// PrintDateTime will print both date & time
+	PrintDateTime bool
+}
+
+// RenderTable is generic function for rendering a slice of structs as a table
+func RenderTable(w io.Writer, slice interface{}, opts TableOptions) {
+	sliceValue := reflect.ValueOf(slice)
+	if sliceValue.Kind() != reflect.Slice {
+		panic("table must be a slice")
```

Comment:
Two questions here: 
* can we avoid panic? 
* Can we provide more data to the result, for better understanding, what data was passed

The same applies to the next panic

Reply:
I think panics are ok here, as they could only happen during development process.
Agree on second point, added additional info what what actually provided.